llm/dyn_ext_server.go

```diff
@@ -258,6 +255,20 @@ func (llm *dynExtServer) Predict(ctx context.Context, predict PredictOpts, fn fu
 					break out
 				}
 
+				switch {
+				case strings.TrimSpace(p.Content) == lastToken:
+					tokenRepeat++
+				default:
+					lastToken = strings.TrimSpace(p.Content)
```

Comment:
Should not trim space since <space>string and string are two separate tokens

Reply:
They are separate tokens, but for the sake of repetition they are the same. It is common for the LLM to output string tokens of differing lengths when stuck in a prediction loop. In that was we still should be able to handle it.